### PR TITLE
[@types/stripe] Adding Issuing types and updating tests

### PR DESCRIPTION
**Pull request by collin-pham**
Please fill in this template.

Adding Issuing Types to @types/stripe. Stripe Documentation here: https://stripe.com/docs/api/issuing/authorizations

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://stripe.com/docs/api/issuing/authorizations

### CODE REVIEW

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -95,7 +96,8 @@ declare class Stripe {
     disputes: Stripe.resources.Disputes;
     events: Stripe.resources.Events;
     invoices: Stripe.resources.Invoices;
-    invoiceItems: Stripe.resources.InvoiceItems;
+	invoiceItems: Stripe.resources.InvoiceItems;
```
Comment (pushplay) [inline types/stripe/index.d.ts]: It looks like you're using hard tabs here.  That needs to be spaces to fit with the rest of that project.

File: types/stripe/index.d.ts [hunk 2]
```diff
@@ -5206,7 +5208,1262 @@ declare namespace Stripe {
...
+				 * Show child attributes
+				 */
+				verification_data: VerificationData;
+
+				/**
+				 * What, if any, digital wallet was used for this authorization. One of apple_pay, google_pay, or samsung_pay.
+				 */
+				wallet_provider: WalletProvider;
+			}
+			interface MerchantData {
```
Comment (pushplay) [inline types/stripe/index.d.ts]: There should be an empty newline between interfaces.

File: types/stripe/index.d.ts [hunk 3]
```diff
@@ -5206,7 +5208,1262 @@ declare namespace Stripe {
...
+				/**
+				 * If the authorization’s is_held_amount_controllable property is true, you may provide this value to control how much to hold for the authorization.
+				 * Must be positive (use decline to decline an authorization request).
+				 */
+				held_amount?: number;
+
+				/**
+				 * Set of key-value pairs that you can attach to an object. This can be useful for storing additional information about the object in a structured format.
+				 */
+				metadata?: IMetadata;
```
Comment (pushplay) [inline types/stripe/index.d.ts]: Options classes should use `IOptionsMetadata` instead of `IMetadata`.

**Review by pushplay — CHANGES_REQUESTED** [review]
Awesome.  Looks like you put a lot of work into this.  And thanks for the tests.  Just a bit of formatting work and fix the `metadata` on Options classes and it's good to go.

File: types/stripe/index.d.ts (hunk 1 above)
Comment (collin-pham) [reply types/stripe/index.d.ts]: @pushplay resolved. Let me know how it looks now.

File: types/stripe/index.d.ts (hunk 3 above)
Comment (collin-pham) [reply types/stripe/index.d.ts]: @pushplay resolved. Let me know how it looks now.

File: types/stripe/index.d.ts (hunk 2 above)
Comment (collin-pham) [reply types/stripe/index.d.ts]: @pushplay resolved. Let me know how it looks now.

**Review by collin-pham — COMMENTED** [review]
@pushplay resolved comments. Let me know how it looks now.

**Review by pushplay — APPROVED** [review]
Quick turnaround!

**Comment by collin-pham** [comment]
@pushplay any idea why this failed checks?

**Comment by pushplay** [comment]
I don't know much about DT's build process but guessing from the logs it's unhappy about what's going on in https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/lowdb/tsconfig.json which isn't your fault.  Not sure what to do about that.

**Comment by collin-pham** [comment]
@sandersn I noticed you are changing some high level behavior in DT. Did checks in this PR fail because of those changes?

**Comment by sandersn** [comment]
Yep, there are about 6 PRs that got caught by it -- I'm restarting them now.

**Comment by collin-pham** [comment]
@sandersn looks like the same lowdb error is still happening after that run. Let me know if I can do anything on my end.

**Comment by sandersn** [comment]
It looks like it's fixed now that I restarted it a couple of times.

**Comment by collin-pham** [comment]
@pushplay looks like this is good to merge?

**Review by pushplay — APPROVED** [review]

**Pull request merged**